components/render/render.h

```diff
@@ -36,6 +36,34 @@ namespace Level
 
 namespace Render
 {
+    // Difference between two points
+    struct Vector
+    {
+      int32_t x;
+      int32_t y;
+
+      Vector operator* (const double &scalar) const
+      {
+        return {static_cast<int32_t> (x * scalar), static_cast<int32_t> (y * scalar)};
+      }
+    };
+
+    // Point in pixels
+    struct Point
+    {
+      int32_t x;
+      int32_t y;
+      Point operator+ (const Vector &v) const { return {x + v.x, y + v.y}; }
+      Point operator- (const Vector &v) const { return {x - v.x, y - v.y}; }
+      Vector operator- (const Point &other) const { return {x - other.x, y - other.y}; }
+    };
+
+    // Tile mesasured in indexes on tile grid
+    struct Tile
```

Comment:
I'd say it would be fine to just use Vector for this. Also, could combine Point and Vector.
And it should probably be in the Misc component, not here. But I agree, using std::pair is nasty